src/app/services/cart/cart.service.ts

```diff
@@ -0,0 +1,76 @@
+import { map, take } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject, combineLatest } from 'rxjs';
+
+import { CartProduct } from '../../models/cart-product.model';
+import { StoreService } from '../store/store.service';
+import { Product } from 'src/app/models/product.model';
+
+@Injectable()
+export class CartService {
+
+  private cart$: BehaviorSubject<CartProduct[]>;
+
+  constructor(private storeService: StoreService) {
+    this.cart$ = new BehaviorSubject<CartProduct[]>([]);
+  }
+
+  addProduct(name: string) {
+      this.cart$.next([...this.cart$.getValue(), { productName: name, quantity: 1 } as CartProduct]);
+   
+  }
+
+  removeProduct(name: string) {
+    this.cart$.next(this.cart$.getValue().filter(cartProduct => cartProduct.productName !== name));
+  }
+
+  updateProductQuantity(name: string, quantity: number) {
+    let cart = this.cart$.getValue();
+    let productToUpdate = cart.find(cartProduct => cartProduct.productName === name);
+
+    if(productToUpdate) {
+      productToUpdate.quantity = quantity;
+    }
+
+    this.cart$.next(cart);
+  }
+
+  getCartSize(): Observable<number> {
+    return this.cart$.pipe(map((cart: CartProduct[]) => {
+      return cart.length;
+    }));
+  }
+
+  isCartProductExist(name: string): Observable<boolean> {
+    return this.cart$.pipe(map((cart: CartProduct[]) => {
+      return cart.some(cartProduct => cartProduct.productName === name);
+    }));
+  }
+
+  getCartProducts(): Observable<CartProduct[]> {
+    return this.cart$.asObservable();
+  }
+
+  getTotalPrice(): Observable<number> {
```

Comment:
this is not the best solution ... cart service should not know store service .. should be independed... get products observable as param in this method or even better solution u can send the combine latest u did in cart component